app/scripts/controllers/sort-network-list.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  BaseControllerV2,
+  RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import { NetworkState } from '@metamask/network-controller';
+
+const controllerName = 'NetworksOrderingController';
+
+/**
+ * The network ID of a network.
+ */
+export type NetworkId = string;
+
+export type NetworksOrderingControllerState = {
+  networksList: NetworkId[] | null;
+};
+
+export type NetworksOrderingControllerupdateNetworksListAction = {
+  type: `${typeof controllerName}:updateNetworksList`;
+  handler: NetworksOrderingController['updateNetworksList'];
+};
+
+export type NetworksOrderingControllerMessengerActions =
+  NetworksOrderingControllerupdateNetworksListAction;
+
+export type NetworksOrderingControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  NetworksOrderingControllerMessengerActions,
+  never,
+  never,
+  never
+>;
+
+const defaultState = {
+  networksList: [],
+};
+
+const metadata = {
+  networksList: {
+    persist: true,
+    anonymous: true,
+  },
+};
+
+export class NetworksOrderingController extends BaseControllerV2<
```

Comment:
Nit: Perhaps this could be called `NetworkOrderController` instead? That would match the other controller names better; usually they're `[noun]Controller` where the noun is the state that the controller is responsible for, network order in this case. Whereas `NetworkOrderingController` is more like `[verb]Controller`.

Reply:
Updated here https://github.com/MetaMask/metamask-extension/pull/21163/commits/759afcb14a045c11dbff282d07bb59d743e07659

Reply:
Hmm, I see that it's still `Networks` instead of `Network`. The extra `s` seems strange to me, i.e. we wouldn't call the ordering of this list the "networks order", we'd call it the "network order". Similar to how we use singular over plural for enums (see https://consensys.slack.com/archives/G1L7H42BT/p1677616875751329)